Minio/RequestExtensions.cs

```diff
@@ -101,19 +100,26 @@ public static Task WrapperPutAsync(this IMinioClient minioClient, string url, St
                 await requestMessageBuilder.ResponseWriter(responseResult.ContentStream, cancellationToken)
                     .ConfigureAwait(false);
 
-            var path = request.RequestUri.LocalPath.TrimStart('/').TrimEnd('/').Split('/');
+            var path = request.RequestUri.LocalPath.TrimStart('/').TrimEnd('/')
+                .Split('/', StringSplitOptions.RemoveEmptyEntries);
             if (responseResult.Response.StatusCode == HttpStatusCode.NotFound)
             {
+                if (request.RequestUri.ToString().Contains("lock", StringComparison.OrdinalIgnoreCase) &&
+                    request.Method == HttpMethod.Get)
+                {
+                    responseResult.Exception = new MissingObjectLockConfigurationException();
+                    return responseResult;
+                }
```

Comment:
Just checking if the request url contains `lock` will also result in a `MissingObjectLockConfigurationException` for files that have lock in the bucket-name or key. It's much safer to actually parse the query string and check for the `object-lock` parameter:
```suggestion
                if (request.Method == HttpMethod.Get)
                {
                    var q = HttpUtility.ParseQueryString(request.RequestUri.Query);
                    if (q.Get("object-lock") != null)
                    {
                        responseResult.Exception = new MissingObjectLockConfigurationException();
                        return responseResult;
                    }
                }

```

Reply:
Done.